src/harness.js

```diff
@@ -234,11 +235,41 @@ class Harness {
       /* this.log respects verbose option, use logFormErrors here */
       console.error(`Error encountered while filling form:`, JSON.stringify(fillResult.errors, null, 2));
     }
+    return fillResult;
+  }
 
+  /**
+   * Loads and fills a contact form,
+   *
+   * @param {string} contactType Type of contact that should be created
+   * @param  {...string[]} answers Provide an array for the answers given on each page. See fillForm for more details.
+   */
+  async fillContactForm(contactType, ...answers) {
+    const fillResult = await this._fillContactForm(contactType, 'create', ...answers);
     this.pushMockedDoc(...fillResult.contacts);
     return fillResult;
   }
 
+  /**
+   * Loads and fills a contact edit form,
+   *
+   * @param {string} contactType Type of contact that should be created
+   * @param  {...string[]} answers Provide an array for the answers given on each page. See fillForm for more details.
+   */
+  async fillContactEditForm(contactType, ...answers) {
+    const assignValues = (target, props) =>
+      Object.entries(props).forEach(([k, v]) => (k !== '_id' && v) && (target[k] = v)); // replace all non empty/null values
+
+    const fillResult = await this._fillContactForm(contactType, 'edit', ...answers);
+    const _contact = this.subject;
+    assignValues(_contact, fillResult.contacts[0]);
+
+    this._state.contacts = this._state.contacts.filter(contact => contact._id !== this.subject._id); // remove the contact
```

Comment:
should we introduce an interface to abstract this out? e.g deleteMockedDoc ?

Reply:
do you actually need to remove it and readd it since the array is shallow (just a list of references). if you assign to the object, it should change automatically in the array which contains it - no?